specification.md

```diff
@@ -2201,6 +2219,169 @@ authenticity of the downloaded file, for example can be the
 The exact signing and verification method is Agent specific. See
 [Code Signing](#code-signing) for recommendations.
 
+## Custom Messages
+
+### Motivation
+
+The OpAMP protocol is intended to cover all aspects of remote management and configuration of Agents. If more
+requirements within this core functionality are identified, it is expected that the protocol will be extended to support
+this additional behavior in future releases. However, there are some use cases that require communication with Agents
+that will not by supported by the OpAMP protocol because they are not within the scope of Agent remote management. The
+CustomMessage allows custom behavior to be implemented between a Server and Agent without having to open another
+connection or define an entirely new protocol.
+
+While interoperability between many Agents and Agent Management Servers is a goal of OpAMP, additional custom
+functionality may be only supported by specific Agents and Servers. Where appropriate, implementers are encouraged to
+document their use of CustomMessage so that support can be added to additional platforms.
+
+### CustomMessage
+
+CustomMessage allows for custom messages to be sent between the Agent and Server. It is supported on ServerToAgent and
+AgentToServer. It requires that the Agent and Server both agree on the contents and encoding of the message. If the type
+is not recognized, the message MUST be ignored and a reply with CustomMessageAck with Status NOT_SUPPORTED MUST be sent.
```

Comment:
For some reason the `MUST be ignored` part of this line seems a bit heavy handed to me. I'm wondering if it's fine to just say `MUST replay with CustomMessageAck with Status NOT_SUPPORTED`. Not sure if we need to explicitly call out the receiver must ignore the message.

Reply:
My intent was to make it clear that it is not an error, but I could modify the sentence.

Reply:
Since there are only three options for a Status I think the fact that it's called out it has to be `NOT_SUPPORTED` rather than `ERROR` conveys this.